### ISSUE TITLE
Loss increasing in joints training of no hand bones data

### ISSUE THREAD

**Issue by puhuajiang** [jasongzy/Make-It-Animatable#10]
Thank you for your excellent work and open-sourcing it. I'm currently using this code to fine-tune my own Mixamo data. However, my data doesn't have the ground truth (GT) of hand bones. I've found that without making any code modifications, training the bw and coarse joints works fine. But when it comes to training the joints part, the evaluation loss shows the situation in the following figure.

<img width="1057" alt="Image" src="https://github.com/user-attachments/assets/9f4d3746-bddd-4e92-b2f1-8ad6b150f84a" />

Regarding the joints connect loss, I think it's because the loss of the hand part is included in the calculation, so the loss keeps increasing. As for the joints loss, I don't quite understand why this overfitting phenomenon occurs. Could you help me identify the problem?
Or, if I want to train Mixamo data without hand bones, what modifications should I make to the code (I hope to use your previous network structure and checkpoints)?

**Comment by jasongzy** [comment]
For the joints_connect loss, you can edit the following lines to exclude finger bones:

https://github.com/jasongzy/Make-It-Animatable/blob/ccbd7d01b72bf69427bb374d60b32c1d23f46fd7/util/dataset_mixamo.py#L180-L189

As for the joints loss, maybe the overfitting problem comes from the causal transformer of the decoding part. Without enough high-quality and versatile data, the causal transformer is prone to overfitting. So you can replace it with an MLP decoder by setting ``--joints_attn_causal 0``.

And if you still want to load the pretrained checkpoint in that case, add this flag: `--resume_strict 0`.

**Comment by puhuajiang** [comment]
Thank you for your response. I'll go ahead and give it a try.

**Comment by puhuajiang** [comment]
@jasongzy Following your advice, I replaced it with an MLP and successfully eliminated the overfitting problem. I really appreciate it. However, I have another question. In the dataset you provided, the number of characters is small, while the number of animations is large. I previously used a casual transformer and increased the number of characters by 1,000. To reduce the dataset size, I reduced the number of animations to around 10. Do you think that reducing the number of animations and increasing the number of characters could degrade the data quality and cause overfitting? How should I balance the number of characters and animations?

**Comment by puhuajiang** [comment]
Thank you for your reply. I will explore this issue.

**Comment by puhuajiang** [comment]
Sorry to bother you again,  I noticed that the output mesh has been globally rotated. May I ask how to unrotate the output mesh after rigging? The pipeline should be: Input -> rotated -> rigging -> unrotated -> output

**Comment by jasongzy** [comment]
The global transformation includes a normalization (scaling & translation) and a rotation. This transformation can be obtained in the preprocess method:

https://github.com/jasongzy/Make-It-Animatable/blob/ccbd7d01b72bf69427bb374d60b32c1d23f46fd7/app.py#L647

You can use `global_transform.inverse().transform_points()` to transform any 3D points (vertices or joints) back to the original coordinates.
